indexer/services/comlink/__tests__/controllers/api/v4/compliance-v2-controller.test.ts

```diff
@@ -322,5 +332,166 @@ describe('ComplianceV2Controller', () => {
       expect(response.status).toEqual(200);
       expect(response.body.status).toEqual(ComplianceStatus.COMPLIANT);
     });
+
+    it('should set status to BLOCKED for ONBOARD action from a restricted country with no existing compliance status', async () => {
+      (Secp256k1.verifySignature as jest.Mock).mockResolvedValueOnce(true);
+      getGeoComplianceReasonSpy.mockReturnValueOnce(ComplianceReason.US_GEO);
+      isRestrictedCountryHeadersSpy.mockReturnValue(true);
+
+      const response: any = await sendRequest({
+        type: RequestMethod.POST,
+        path: '/v4/compliance/geoblock',
+        body: {
+          address: testConstants.defaultAddress,
+          message: 'Test message',
+          action: ComplianceAction.ONBOARD,
+          signedMessage: sha256(Buffer.from('msg')),
+          pubkey: new Uint8Array([/* public key bytes */]),
+          timestamp: 1620000000, // Valid timestamp
+        },
+        expectedStatus: 200,
+      });
+
+      const data: ComplianceStatusFromDatabase[] = await ComplianceStatusTable.findAll({}, [], {});
+      expect(data).toHaveLength(1);
+      expect(data[0]).toEqual(expect.objectContaining({
+        address: testConstants.defaultAddress,
+        status: ComplianceStatus.BLOCKED,
+        reason: ComplianceReason.US_GEO,
+      }));
+
+      expect(response.body.status).toEqual(ComplianceStatus.BLOCKED);
+      expect(response.body.reason).toEqual(ComplianceReason.US_GEO);
+    });
+
+    it('should set status to FIRST_STRIKE for CONNECT action from a restricted country with no existing compliance status', async () => {
+      (Secp256k1.verifySignature as jest.Mock).mockResolvedValueOnce(true);
+      getGeoComplianceReasonSpy.mockReturnValueOnce(ComplianceReason.US_GEO);
+      isRestrictedCountryHeadersSpy.mockReturnValue(true);
+
+      const response: any = await sendRequest({
+        type: RequestMethod.POST,
+        path: '/v4/compliance/geoblock',
+        body: {
+          address: testConstants.defaultAddress,
+          message: 'Test message',
+          action: ComplianceAction.CONNECT,
+          signedMessage: sha256(Buffer.from('msg')),
+          pubkey: new Uint8Array([/* public key bytes */]),
+          timestamp: 1620000000, // Valid timestamp
+        },
```

Comment:
nit: Pull out this body into a separate object so we don't have to repeat this per test-case?

Reply:
done